README.md

```diff
@@ -1,16 +1,121 @@
-[![TeamCity](https://teamcity.int.avast.com/app/rest/builds/buildType:backends_JvmLibs_GrpcUtils_BuildTest,branch:<default>/statusIcon)](https://teamcity.int.avast.com/project.html?projectId=backends_JvmLibs_GrpcUtils)
+# gRPC JSON Bridge
 
-# gRPC utils
+[![Build Status](https://travis-ci.org/avast/grpc-json-bridge.svg?branch=master)](https://travis-ci.org/avast/grpc-json-bridge)
+[![Download](https://api.bintray.com/packages/avast/maven/grpc-json-bridge/images/download.svg) ](https://bintray.com/avast/maven/grpc-json-bridge/_latestVersion)
 
-Set of classes to help us work with gRPC
+This library makes possible to receive a JSON encoded request to a gRPC service. It provides an implementation-agnostic module for mapping to
+your favorite HTTP server as well as few implementations for direct usage in some well-known HTTP servers.
 
-## GrpcJsonWrapper
+It uses Scala macros for creating mapping between runtime-provided service and method names to pregenerated Java gRPC classes. In case you
+don't want to use _plain Java API_ you can easily use it together with [Cactus](https://github.com/avast/cactus).
 
-Provide HTTP JSON API by gRPC proto contract.
+There are several modules:
+1. core - for basic implementation-agnostic usage
+1. [http4s](http4s) - integration with [http4s](https://http4s.org/) webserver
+1. [akka-http](akka-http) - integration with [Akka Http](https://doc.akka.io/docs/akka-http/current/server-side/index.html) webserver
 
-### Setup and start http server
-For example we have proto file  with definition of `ControllerService` and gRPC service implementation `ControllerApiGrpcService`.
+The created [`GrpcJsonBridge`](core/src/main/scala/com/avast/grpc/jsonbridge/GrpcJsonBridge.scala) exposes not only the methods itself but
+also provides their list to make possible to implement an _info_ endpoint (which is already implemented in server-agnostic implementations).
 
-    val wrapper = new InProcessJsonWrapperBuilder(classOf[ControllerServiceGrpc], new ControllerApiGrpcService(flowService, stateStore))
-    val httpServer = new GrpcYapWrapperServerBuilder("localhost", 8080, wrapper).start()
+Recommended URL scheme for exposing the service (and the one used in provided implementations) is `/$SERVICENAME/$METHOD` name and the http
```

Comment:
I don't like the word _scheme_ here, since it has a specific meaning in terms of URIs/URLs. URL pattern?

Reply:
OK